src/cloud/components/organisms/EditorLayout/NewDocContextMenu.tsx

```diff
@@ -0,0 +1,195 @@
+import {
+  mdiAccountCircleOutline,
+  mdiAccountMultiple,
+  mdiClockOutline,
+  mdiContentSaveOutline,
+} from '@mdi/js'
+import React, { useMemo, useState } from 'react'
+import Flexbox from '../../../../shared/components/atoms/Flexbox'
+import MetadataContainer from '../../../../shared/components/organisms/MetadataContainer'
+import MetadataContainerBreak from '../../../../shared/components/organisms/MetadataContainer/atoms/MetadataContainerBreak'
+import MetadataContainerRow from '../../../../shared/components/organisms/MetadataContainer/molecules/MetadataContainerRow'
+import {
+  SerializedDoc,
+  SerializedDocWithBookmark,
+} from '../../../interfaces/db/doc'
+import { SerializedRevision } from '../../../interfaces/db/revision'
+import { SerializedTeam } from '../../../interfaces/db/team'
+import { SerializedUser } from '../../../interfaces/db/user'
+import { getFormattedDateTime } from '../../../lib/date'
+import { useI18n } from '../../../lib/hooks/useI18n'
+import { lngKeys } from '../../../lib/i18n/types'
+import { usePage } from '../../../lib/stores/pageStore'
+import SmallButton from '../../atoms/SmallButton'
+import UserIcon from '../../atoms/UserIcon'
+import BackLinksList from './molecules/BackLinksList'
+import ContextMenuClose from './molecules/ContextMenuClose'
+import DocContextMenuActions from './molecules/DocContextMenuActions'
+
+interface DocContextMenuProps {
+  currentDoc: SerializedDocWithBookmark
+  contributors: SerializedUser[]
+  backLinks: SerializedDoc[]
+  team: SerializedTeam
+  editorRef?: React.MutableRefObject<CodeMirror.Editor | null>
+  restoreRevision?: (revision: SerializedRevision) => void
+}
+
+const DocContextMenu = ({
+  team,
+  currentDoc,
+  contributors,
+  backLinks,
+  editorRef,
+}: DocContextMenuProps) => {
+  const { permissions = [], currentUserIsCoreMember } = usePage()
```

Comment:
Probably doesn't need to be changes, but this data can probably be passed as props rather than using the hook. `team,` `currentDoc` etc is so it feels a bit out of place.
Maybe worth taking a look at if/when we review how we handle API data, as here it seems to actually only be used to ensure most recent version of users, which probably should be done at a higher level.